### DISCUSSION TITLE
U and W on CNC Lathe

### DISCUSSION THREAD

**Discussion by Reprosense** [grblHAL/core#398] [General]
hi everyone , is there or will be any incremental programming for lathe which will using an U and W axis , beside of X and Z for absolute ? , because from what i get in GRBL they just change an UVW for ABC axis

**Comment by terjeio** [comment]
You mean [like this](https://www.cnctrainingcentre.com/haas-turn/u-and-w-on-a-cnc-lathe-incremental-programming/)? If so it is not currently supported but may not be too hard to implement.

**Comment by Reprosense** [comment]
yea , okay thanks for the information terje

**Reply by terjeio** [reply]
FYI UVW support in lathe mode is available as a compile time option in the [latest commit.](https://github.com/grblHAL/core/blob/master/changelog.md#20231210)

**Reply by Reprosense** [reply]
thanks terje

**Comment by LEOOZERAN** [comment]
TELL US. How to connect stepper motors to a lathe for thread cutting. What kind of stepper motor is installed in the chuck of a lathe?

**Reply by terjeio** [reply]
First - do not ask unrelated questions in existing discussions, open a new one instead.

Using a stepper motor to drive the spindle has recently become an option but is as of now untested. What kind of stepper to use will depend on the size of the lathe and the materials it is to be used for? A internet search may return some useful results?
FYI my lathe has a 3 phase spindle motor controlled by a VFD.
